service/src/main/java/bio/terra/pipelines/db/repositories/JobsRepository.java

```diff
@@ -0,0 +1,15 @@
+package bio.terra.pipelines.db.repositories;
+
+import bio.terra.pipelines.db.entities.DbJob;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.data.repository.CrudRepository;
+
+public interface JobsRepository extends CrudRepository<DbJob, Long> {
+  List<DbJob> findAllByPipelineIdAndUserId(String pipelineId, String userId);
+
+  Optional<DbJob> findJobByPipelineIdAndUserIdAndJobId(
+      String pipelineId, String userId, String jobId);
+
+  Optional<DbJob> findJobByJobId(String jobId);
```

Comment:
Can't you use `CrudRepository.findById()` for this?

Reply:
i didnt want to use this since I planned on changing the `@Id` annotated column for this table in the future and the `findByJobId` will always work.